update for Bevy 0.13

tzemanovic opened the pull request:
hi, thanks for the crate :) 

closes #23

I tried to update for 0.12 earlier and it was tricky as there was no released iced version with a matching winit version. Both iced and bevy have a new release which can be used together again

tasgon commented on Cargo.toml:
```diff
@@ -13,21 +13,23 @@ documentation = "https://docs.rs/bevy_iced"
-bevy_render = "0.11"
-bevy_utils = "0.11"
-bevy_window = "0.11"
+# NOTE: temp using "main" branch until a release with wgpu version matching iced
```
Is this comment (and the one below) still relevant?

tzemanovic replied:
ah, no. I'll remove them

tasgon commented on .gitignore:
```diff
@@ -1,5 +1,4 @@
 /target
-Cargo.lock
```
This is a library crate. Why are we no longer ignoring `Cargo.lock`?

tzemanovic replied:
the guidance has changed on this recently https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html but if you prefer to keep it out I can fix it up

tasgon replied:
Alright, I'll think about this.

tasgon commented:
Sorry for the delay. I reviewed the PR and corrected a few bugs (fixed some of the examples and text inputs), but there's still two issues I'm seeing:

- I'm noticing flickering whenever I edit the text in the `interactive` example. This could have something to do with how I fixed it, though.
- The `fonts` example doesn't seem to be actually using the custom font. Not just that, but `default_text_size` is also being ignored. Not sure if this is a bug with this crate or iced, but AFAIK iced's font handling behavior changed drastically in 0.12, so we might be doing something here.

I'm about ready to merge this, but would you mind taking a look at these two points first?

tzemanovic commented:
> Sorry for the delay. I reviewed the PR and corrected a few bugs (fixed some of the examples and text inputs), but there's still two issues I'm seeing:
> 
>     * I'm noticing flickering whenever I edit the text in the `interactive` example. This could have something to do with how I fixed it, though.
> 
>     * The `fonts` example doesn't seem to be actually using the custom font. Not just that, but `default_text_size` is also being ignored. Not sure if this is a bug with this crate or iced, but AFAIK iced's font handling behavior changed drastically in 0.12, so we might be doing something here.
> 
> 
> I'm about ready to merge this, but would you mind taking a look at these two points first?

Thanks for fixes! I'm also seeing the flickering and the issue with font settings. The issue with flicker is already present in the first commit (https://github.com/tasgon/bevy_iced/pull/24/commits/7b3415bb37822ca018c669425ea88fcc837dd348). I'll look into these and set this back to draft in the meantime

tzemanovic commented:
fixed the font settings in [19aaa6d](https://github.com/tasgon/bevy_iced/pull/24/commits/19aaa6d416f4a47fafcc64173063304820225a9a)

tzemanovic commented:
I did a bit of digging and found what’s happening with the flickering - see https://github.com/iced-rs/iced/pull/2058

tasgon commented:
Anything else still blocking this PR from being merged? Because if the flicker is the only thing, I'm thinking we should just merge this in now so people can at least be able to use the library instead of not at all.

n1ght-hunter commented on src/conversions.rs:
```diff
@@ -183,6 +331,9 @@ pub fn mouse_button(button: MouseButton) -> iced_core::mouse::Button {
         MouseButton::Right => Button::Right,
         MouseButton::Middle => Button::Middle,
         MouseButton::Other(val) => Button::Other(val),
+        // TODO: We need https://github.com/iced-rs/iced/pull/2169
```
this has been merged
https://github.com/iced-rs/iced/pull/2169

tzemanovic replied:
fixed in [e003361](https://github.com/tasgon/bevy_iced/pull/24/commits/e003361034712798a71a5ffa0ea921dc4fe634e6)

tzemanovic commented:
> Anything else still blocking this PR from being merged? Because if this flicker is the only thing, I'm thinking we should just merge this in now so people can at least be able to use the library instead of not at all.

Let make a temp fork of iced with disabled cache to workaround this and open an issue here to track it, then I think we can merge. The issue is even worse with a secure input, so I don't think we should merge without it

tzemanovic commented:
switched to cache-less text_input fork (https://github.com/tzemanovic/iced/commit/ac826cf5f72a0598bbe986fd6da9f970ef91ab98 on top of iced v0.12.1)

tasgon commented:
Sorry for the delay, but I generally prefer it if the `master` branch is publishable to Crates.io, and relying on a fork of `iced` would make that impossible (unless the fork is also published). If it's only the `TextInput` component that's causing problems, do you think it would be possible to vendor in the component into `bevy_iced`?

tasgon commented:
For now, I've switched back to using mainline. If users want to use the custom fork, I'm pretty sure they can do that using [`[patch]`](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html).
